api/app/controllers/spree/api/v2/storefront/taxons_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class TaxonsController < ::Spree::Api::V2::BaseController
+          def index
+            @taxons = Spree::Taxon.includes(:children).accessible_by(current_ability, :read)
+            @taxons = dependencies[:taxons_finder].new(@taxons, params).execute
+
+            render json: serialize_taxons(@taxons), status: 200
+          end
+
+          def show
+            @taxon = Spree::Taxon.find_by(id: params[:id])
+
+            if @taxon
+              render json: serialize_taxon(@taxon), status: 200
+            else
+              render json: { error: "Taxon doesn't exist" }, status: 422
```

Comment:
This probably should be status 404. 422 is more for processing entity and not just fetching it.

http://jsonapi.org/format/#fetching-resources-responses-404

We should discuss error format(we need i18n here definitely)